src/memray/_memray/snapshot.h

```diff
@@ -320,6 +320,64 @@ class HighWaterMarkAggregator
     reduced_snapshot_map_t getAllocations(bool merge_threads, bool stop_at_high_water_mark) const;
 };
 
+struct AllocationLifetime
+{
+    size_t allocatedBeforeSnapshot;
+    size_t deallocatedBeforeSnapshot;  // SIZE_MAX if never deallocated
+    HighWaterMarkLocationKey key;
```

Comment:
Now that I am reviewing this, `HighWaterMarkLocationKey` is now a slightly confusing name. I opened an issue about the whole struct:

https://github.com/bloomberg/memray/issues/321